vm.go

```diff
@@ -617,17 +618,22 @@ func (vm *vm) run() {
 				ctx = context.Background()
 			}
 
-			if waitErr := vm.r.limiter.WaitN(ctx, vm.r.limiterTicksLeft); waitErr != nil {
-				if vm.r.vm.ctx == nil {
-					panic(waitErr)
-				}
-				if ctxErr := vm.r.vm.ctx.Err(); ctxErr != nil {
-					panic(ctxErr)
-				}
-				if strings.Contains(waitErr.Error(), "would exceed") {
-					panic(context.DeadlineExceeded)
+			now := time.Now()
+			r := vm.r.limiter.ReserveN(now, vm.r.limiterTicksLeft)
+			if !r.OK() {
+				panic("failed to make reservation")
+			}
+
+			delay := r.DelayFrom(now)
+			if delay > 0 {
+				vm.r.limiterWaitCount++
+				vm.r.limiterWaitTotalTime += delay.Nanoseconds()
```

Comment:
[nit] thoughts on changing `limiterWaitTotalTime` to be of type `time.Duration`? That makes what's returned by `LimiterWaitTotalTime` a little more explicit without having to read the docstring.

Reply:
Ooo I like that! Great idea